### PR TITLE
first-spring-task-solution

### PR DESCRIPTION
**Pull request by SvNadorozhnyi**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    </licenses>
+    <developers>
+        <developer/>
+    </developers>
+    <scm>
+        <connection/>
+        <developerConnection/>
+        <tag/>
+        <url/>
+    </scm>
```
Comment (liliia-ponomarenko) [inline pom.xml]: Let's remove unused tags

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    <properties>
+        <java.version>17</java.version>
```
Comment (liliia-ponomarenko) [inline pom.xml:17]: Let's add checkstyle plugin

File: src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```
Comment (liliia-ponomarenko) [inline src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java:17]: this annotation is redundant. anyway let's add Lombok and use RequiredArgsConstructor annotation

File: src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert book: " + book, e);
```
Comment (liliia-ponomarenko) [inline src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java:36]: Better use DataProcessingException (but you remove this method during this module, so you can leave it)

File: src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.List;
+import org.example.bookstore.model.Book;
+import org.example.bookstore.repository.BookRepository;
+import org.example.bookstore.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    @Autowired
```
Comment (liliia-ponomarenko) [inline src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java]: use RequiredArgsConstructor

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -0,0 +1,8 @@
+spring.application.name=BookStore
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+spring.datasource.url=jdbc:mysql://localhost:3306/test?serverTimezone=UTC
+spring.datasource.username=root
+spring.datasource.password=userpass
+
+spring.jpa.hibernate.ddl-auto=create-drop
+spring.jpa.show-sql=true
```
Comment (liliia-ponomarenko) [inline src/main/resources/application.properties]: Add one empty line at the end of the file (checkstyle plugin helps you;)

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution ;)

**Review by sarakhmen — APPROVED** [review]
lgtm!

**Pull request merged**